handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -239,34 +239,36 @@ protected void deallocate() {
 
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
```

Comment:
The instance is [published](https://github.com/netty/netty/pull/6955/files#diff-cfd417915faad637e4278b54adafd9baR229) before the constructor completes.